tests/test_source_sampling.py

```diff
@@ -38,21 +38,42 @@ def test_single_tet_tag_names_map():
     e_bounds = np.array([0, 0.5, 1.0])
     m.bias = IMeshTag(2, float)
     m.bias[:] = [[1.0, 2.0], [3.0, 3.0]]
-    m.mesh.save("sampling_mesh.h5m")
     filename = "sampling_mesh.h5m"
+    m.mesh.save(filename)
 
     # right condition
     tag_names = {"src_tag_name": "src"}
     e_bounds = np.array([0, 1])
     sampler = Sampler(filename, tag_names, e_bounds, 0)
+
     # src_tag_name not given
     tag_names = {}
     assert_raises(ValueError, Sampler, filename, tag_names, e_bounds, 0)
     assert_raises(ValueError, Sampler, filename, tag_names, e_bounds, 1)
+
     # bias_tag_name not given
     tag_names = {"src_tag_name": "src"}
     assert_raises(ValueError, Sampler, filename, tag_names, e_bounds, 2)
 
+    # subvoxel r2s source.h5m used for r2s calculation
+    cell_fracs = np.zeros(2, dtype=[('idx', np.int64),
+                                    ('cell', np.int64),
+                                    ('vol_frac', np.float64),
+                                    ('rel_error', np.float64)])
+    cell_fracs[:] = [(0, 11, 1.0, 0.0), (1, 11, 1.0, 0.0)]
+    m.tag_cell_fracs(cell_fracs)
+    m.mesh.save(filename)
+    tag_names = {"src_tag_name": "src",
+                 "cell_number_tag_name": "cell_number",
+                 "cell_fracs_tag_name": "cell_fracs"}
+    assert_raises(ValueError, Sampler, filename, tag_names, e_bounds, 0)
```

Comment:
What are 0, 1 and 2 here in these `assert_raises` tests?

Reply:
I will change these 0, 1, 2 ... to something like DEFAUL_ANALOG, DEFAULT_UNIFORM, ... in another PR. As these are two independent changes, better to be in different PRs.

Reply:
Maybe it would be easiest to update this PR...